sympy/solvers/solvers.py

```diff
@@ -378,7 +380,8 @@ def checksol(f, symbol, sol=None, **flags):
     # TODO: improve solution testing
 
 
-def solve(f, *symbols, **flags):
+def solve(f, *symbols, **flags) -> Union[Basic, dict, List[Union[dict, Any]],
+                                         Any]:
```

Comment:
Curious, does Union[..., Any] actually provide any benefit? From a typing perspective, this type is no different from just Any, no?

Reply:
In VS Code at least, it means that the type is often the thing other than Any, so it'll provide hints for that thing.

Reply:
I'd imagine that officially it may not be any different to just putting Any there